opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -1037,21 +1138,27 @@ class WellModelGhostLastMatrixAdapter : public Dune::AssembledLinearOperator<M,X
         mutable bool converged_;
         std::any parallelInformation_;
 
-        std::unique_ptr<Matrix> matrix_;
+        std::shared_ptr<Matrix> matrix_;
```

Comment:
This is supposed to be a pointer to the simulator's assembly matrix, so should it not just be a naked pointer? Using unique_ptr or shared_ptr implies that this class participates in ownership.

Reply:
Indeed. That seems historical baggage. Will fix,